### PR TITLE
Turning Sieve benchmarks into Enso benchmarks

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Having the benchmarks in Enso will make it easier to rewrite certain parts of the algorithm to other languages.

### Important Notes

Execute as
```bash
sbt:enso> runEngineDistribution --run test/Benchmarks Sieve
```
or as
```bash
sbt:enso> std-benchmarks/benchOnly Sieve
```

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), Enso
      style guides.
- All code has been tested:
  - [x] Benchmarks can be executed
  - [x] New benchmarks provide similar results as old ones

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
Current results on my computer:
```
[info] Benchmark            Mode  Cnt    Score   Error  Units
[info] Sieve.Ascribed       avgt       291.542          ms/op
[info] Sieve.Original       avgt       289.745          ms/op
[info] Sieve.Without_Types  avgt       270.837          ms/op
```

File: test/Benchmarks/src/Sieve/Sieve_Java_Script.enso [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        next() {
+            this.n++
+            return this
+        }
+    }
+    return new Gen()
+
+compute_nth_prime_natural_in_js = Sieve_Without_Types.compute_nth_prime gen=natural_in_js
```
Comment (JaroslavTulach) [inline test/Benchmarks/src/Sieve/Sieve_Java_Script.enso:22]: These JavaScript benchmarks work with:
```
sbt:enso> runEngineDistribution --run test/Benchmarks Sieve
```
but not with `std-benchmarks/benchOnly Sieve`. Please help, @Akirathan!

**Comment by JaroslavTulach** [comment]
[a2acce6](https://github.com/enso-org/enso/pull/8475/commits/a2acce64b20cecfe27ef627a9c977e25907ce13b) adds JavaScript benchmarks. They however seem to show the price we pay for JavaScript interop is high! 
```
[info] Sieve.Ascribed             avgt    3   268.487 ±   60.973  ms/op
[info] Sieve.Java_Script_All      avgt    3   888.633 ±  276.752  ms/op
[info] Sieve.Java_Script_Filter   avgt    3  1831.328 ± 4293.687  ms/op
[info] Sieve.Java_Script_Natural  avgt    3  2152.961 ±  804.918  ms/op
[info] Sieve.Original             avgt    3   278.269 ±  104.952  ms/op
[info] Sieve.Without_Types        avgt    3   274.963 ±   96.071  ms/op
```
`Java_Script_Filter` and `Java_Script_Natural` pay a price for interop - clearly not done at the fastest possible manner. But why `Java_Script_All` is slow remains a mystery - standalone benchmarks show JavaScript is faster than Enso.

Running [official benchmarks](https://github.com/enso-org/enso/actions/runs/7116709989) and [once again](https://github.com/enso-org/enso/actions/runs/7125369080).

obrazek

The picture confirms (not surprisingly) that the "Enso Sieve Benchmarks" run at the same speed as the "JMH Sieve Benchmarks". E.g. we can integrate and use/watch the new benchmarks.

File: lib/scala/bench-processor/src/main/java/org/enso/benchmarks/processor/BenchProcessor.java [hunk 2]
```diff
@@ -231,7 +231,7 @@ private void generateClassForGroup(
         "      throw new IllegalStateException(\"Project root directory does not exist or cannot be read: \" + Objects.toString(projectRootDir));");
     out.println("    }");
     out.println("    File languageHomeOverride = Utils.findLanguageHomeOverride();");
-    out.println("    var ctx = Context.newBuilder(LanguageInfo.ID)");
+    out.println("    var ctx = Context.newBuilder()");
```
Comment (JaroslavTulach) [inline lib/scala/bench-processor/src/main/java/org/enso/benchmarks/processor/BenchProcessor.java:234]: I thought it was going to be a classpath issue, not "restricted to Enso language" issue! Thank you for your help, @Akirathan!

**Review by radeusgd — APPROVED** [review]

**Comment by radeusgd** [comment]
> But why `Java_Script_All` is slow remains a mystery - standalone benchmarks show JavaScript is faster than Enso.

Hmm this is very curious indeed 🤔

**Review by hubertp — APPROVED** [review]
Does it make sense to keep JMH benchmark as well - the more the merrier and gives us some reference?

**Comment by JaroslavTulach** [comment]
Official results are here. `Java_Script_File` takes `70ms` - that's the goal we'd like to reach one day... `Original`, `Ascribed` and `Without_Types` take `400+ms` - they are purely written in Enso. E.g. Enso is 6 times slower than JavaScript. The other `Java_Script_...` benchmark somehow mix Enso and JavaScript - they are pretty slow - e.g. certainly there is a cost of crossing the polyglot boundary between Enso & JavaScript.

```
Sieve.Java_Script_File                                                avgt    3    70.976 ±      6.440  ms/op

Sieve.Ascribed                                                        avgt    3   440.056 ±     54.563  ms/op
Sieve.Original                                                        avgt    3   412.924 ±     30.009  ms/op
Sieve.Without_Types                                                   avgt    3   405.650 ±      6.148  ms/op

Sieve.Java_Script_All                                                 avgt    3  1123.099 ±    391.940  ms/op
Sieve.Java_Script_Filter                                              avgt    3  2609.217 ±   3943.999  ms/op
Sieve.Java_Script_Natural                                             avgt    3  3182.236 ±   7700.466  ms/op
```

**Comment by JaroslavTulach** [comment]
> Does it make sense to keep JMH benchmark as well - the more the merrier and gives us some reference?

Probably not. The results are the same with the Enso benchmarking infrastructure `400-420`ms.

**Pull request merged**